File: comp/A_Satisfying_Constraints.cpp
#include <iostream>
#include <vector>

using namespace std;

int main() {
    int cases;
    cin >> cases;

    vector<int> result;

    for(int i = 0; i < cases; i++) {
        int subcases;
        cin >> subcases;

        int bottom = 0;
        int top = 2147483647;
        vector<int> blacklist;

        for(int j = 0; j < subcases; j++){
            int constraint, value;
            cin >> constraint >> value;
            if (constraint == 1) {
                bottom = max(bottom, value);
            } else if (constraint == 2) {
                top = min(top, value);
            } else {
                blacklist.push_back(value);
            }
        }

        int excluded = 0;
        for (int i : blacklist) {
            if (i >= bottom && i <= top)
                excluded++;
        }

        result.push_back(max(top - bottom - excluded + 1, 0));
    }

    for (int i : result) {
        cout << i << endl;
    }
}

File: comp/A_Satisfying_Constraints.py
cases = int(input())
result = []
for i in range(cases):
    constraint_count = int(input())
    bottom = 0
    top = 10_000_000_000
    blacklist = set()

    for j in range(constraint_count):
        constraint = input().split()
        value = int(constraint[1])

        if constraint[0] == '1':
            bottom = max(bottom, value)
        elif constraint[0] == '2':
            top = min(top, value)
        else:
            blacklist.add(value)
    possible_values = list(range(bottom, top + 1))
    blacklist = [x for x in blacklist if x in possible_values]
    result.append(len(possible_values) - len(blacklist))
for i in result:
    print(i)